From d74920e2a7c413ef40eed72f9cf287cf6fbd5fb8 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Thu, 20 May 2021 14:56:46 +0200
Subject: [PATCH 1/2] EDMA: Fix NAPI packet counting

There is a bug in the NAPI packet counting that will
cause NAPI over budget warnings.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 hal/dp_ops/edma_dp/edma_v1/edma_tx_rx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/hal/dp_ops/edma_dp/edma_v1/edma_tx_rx.c
+++ b/hal/dp_ops/edma_dp/edma_v1/edma_tx_rx.c
@@ -459,12 +459,12 @@ int edma_napi(struct napi_struct *napi,
 
 	for (i = 0; i < ehw->txcmpl_rings; i++) {
 		txcmpl_ring = &ehw->txcmpl_ring[i];
-		work_done += edma_clean_tx(ehw, txcmpl_ring);
+		edma_clean_tx(ehw, txcmpl_ring);
 	}
 
 	for (i = 0; i < ehw->rxfill_rings; i++) {
 		rxfill_ring = &ehw->rxfill_ring[i];
-		work_done += edma_alloc_rx_buffer(ehw, rxfill_ring);
+		edma_alloc_rx_buffer(ehw, rxfill_ring);
 	}
 
 	/*
